src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,48 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
-    public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+
+    public static String getSalaryInfo(String[] names, String[] data,
+                                       String dateFrom, String dateTo) {
+        StringBuilder result = new StringBuilder();
+        result.append(String.format("Report for period %s - %s"
+                + System.lineSeparator(), dateFrom, dateTo));
+        String rowTemplate = "%s - %s";
+        for (int i = 0; i < names.length; i++) {
+            int personTotalSalary = 0;
+            for (int j = 0; j < data.length; j++) {
+                String[] parsedDataRow = parseDataRow(data[j]);
+                DateTimeFormatter formatter
+                        = DateTimeFormatter.ofPattern("d.MM.yyyy");
+                if (LocalDate.parse(parsedDataRow[0], formatter).isAfter(
+                        LocalDate.parse(dateTo, formatter)) || LocalDate.parse(
+                        parsedDataRow[0], formatter).isBefore(
+                        LocalDate.parse(dateFrom, formatter))) {
+                    continue;
+                }
+                if (!names[i].equals(parsedDataRow[1])) {
+                    continue;
+                }
+                personTotalSalary += calculateRowSalary(parsedDataRow[2], parsedDataRow[3]);
```

Comment:
```suggestion
                if (names[i].equals(parsedDataRow[1])) {
                  String[] parsedDataRow = parseDataRow(data[j]);
                  DateTimeFormatter formatter
                          = DateTimeFormatter.ofPattern("d.MM.yyyy");
                  if (LocalDate.parse(parsedDataRow[0], formatter).isAfter(
                          LocalDate.parse(dateTo, formatter)) || LocalDate.parse(
                          parsedDataRow[0], formatter).isBefore(
                          LocalDate.parse(dateFrom, formatter))) {
                      continue;
                  }
                  personTotalSalary += calculateRowSalary(parsedDataRow[2], parsedDataRow[3]);
              }    
```
check if you even need to do some operations or calculations. In this case, you are doing a lot of unnecessary calculations but then doing continue

also avoid magic numbers ✨  use constants